src/OhadSoft.AzureLetsEncrypt.Renewal.WebJob.Tests/WebJob/AppSettingsTests.cs

```diff
@@ -171,6 +172,12 @@ public void TestInvalidEmail()
             AssertInvalidConfig(EmailKeySuffix, WebApp1, "mail@", "email");
         }
 
+        [TestMethod]
+        public void TestInvalidFromEmail()
+        {
+            Assert.ThrowsException<ArgumentException>(() => new SendGridNotifier(string.Empty, "@address"));
```

Comment:
I struggle with tests, and this may have been a shortcut. I'm trying to understand how to use AssertInvalidConfig here.

Since the from address is not part of the RenewalParameters (in fact not part of any parameters class like SendGridApiKey), it seems like AssertInvalidConfig won't work because that method ultimately expects to encounter errors in Renew() and NotifyAsync(), both of which only take a RenewalParameters argument.

Is that accurate?

Can you advise on a good strategy for testing here? (or point me in the right direction if I'm missing something entirely :D ). I am wondering if the SendGridApiKey setting isn't tested for the same reason

Reply:
@ohadschn tagging you here since I've been spamming comments everywhere ....

Reply:
@ohadschn quick ping...should i remove these particular tests or are there alternatives i ought to use?

Reply:
im not sure why I never just put fromEmail on a webapp config (the same way the to address is). that is updated and fromEmail is now tested via AssertInvalidConfig (this also updated CLI to include --fromEmail)